File: cplusplus/design_pattern/MomentoProj/CareTaker.cpp
#include "stdafx.h"
#include "CareTaker.h"


CareTaker::CareTaker()
{
}


CareTaker::~CareTaker()
{
}


CareTaker::CareTaker(std::shared_ptr<Momento> momento) : momento(momento)
{
}


void CareTaker::saveMomento(std::shared_ptr<Momento> momento)
{
    this->momento = momento;
}


std::shared_ptr<Momento> CareTaker::returnMomento()
{
    return momento;
}

File: Java/pattern/Momento/src/com/pattern/CareTaker.java
package com.pattern;

public class CareTaker {
	private Momento momento;
	
	public void saveMomento(Momento momento) {
		this.momento = momento;
	}
	
	public Momento returnMomento() {
		return momento;
	}
	
	public CareTaker(Momento momento) {
		this.momento = momento;
	}

}
